corehq/apps/userreports/data_source_providers.py

```diff
@@ -6,21 +6,29 @@
 
 
 class DataSourceProvider(six.with_metaclass(ABCMeta, object)):
+
     @abstractmethod
-    def get_data_sources(self):
+    def get_all_data_sources(self):
         pass
 
+    def get_data_sources(self, referenced_doc_type=None):
+        sources = self.get_all_data_sources()
+        if self.referenced_doc_type:
```

Comment:
I think this will always evaluate to `False`. Did you mean:

```suggestion
        if referenced_doc_type:
```